app/views/posts/index.html.erb

```diff
@@ -1 +1,63 @@
-<h1>Here is a list of posts with all comments for a given user</h1>
\ No newline at end of file
+
+<section class="container">
+    <div class="user">
+         <div class="user-photo">
+            <img src="<%= @user.photo %>" alt="User Photo">
+        </div>
+
+        <div class="user-info">
+            <%= link_to user_url(@user), class: 'name' do %>
+                <h2><%= @user.name %></h2>
+            <% end %>
+
+            <div class="user-posts">
+                <p>Number of Posts:
+                    <% if @user.posts_counter %>
+                        <%= @user.posts_counter %>
+                    <% else %>
+                        0
+                    <% end %>
+                </p>
+            </div>
+        </div>
+    </div>
+
+    <% @posts.each do |post| %>
+        <div class = "post">
+           <%= link_to user_post_url(user_id: @user.id, id: post.id), class: 'post-link' do %>
+             <h3>Post #<%= @user.posts.index(post) + 1 %></h3>
+           <% end %>
+           <p><%= post.text %></p>
+            <div class="count">
+               <span>Comments:
+                <% if post.comments_counter %>
+                    <%= post.comments_counter %>
+                    <% else %>
+                    0
+                    <% end %>
+                </span>
+               <span>Likes:
+                <% if post.likes_counter %>
+                <%= post.likes_counter %>
+                    <% else %>
+                    0
+                    <% end %>
+                </span>
+            </div>
+        </div>
+
+        <div class="post-comments">
+            <% post.comments.each do |comment| %>
+                <div class="comment">
+                    <span class="comment-author"><%= comment.user.name %>:</span>
+                    <span><%= comment.text %></span>
+                </div>
+            <% end %>
+            </div>
```

Comment:
- [ ] On the posts index page, kindly display only 5 recent comments on each post. You can use the custom method you created in your post model. `post.recent_comments`